### ISSUE TITLE
kubectl logs container arg should be -c

### ISSUE THREAD

**Issue by thockin** [GoogleCloudPlatform/kubernetes#10372]
Today we have:

```
kubectl exec pod [-c container] command
```
and

```
kubectl logs pod [container]
```

In both cases we try to guess the container.  we should use the same notation in both cases.  Either pod.container or pod/container or pod -c container.  If I can't keep it straight, neither can users.

**Comment by bgrant0607** [comment]
See #8448 and #8604. Both take `-c container`. Did you do help or look at the docs? Or did you just try exec without `-c`?

**Comment by thockin** [comment]
```
$ k help logs
Print the logs for a container in a pod. If the pod has only one container,
the container name is optional.

Usage:
  kubectl logs [-f] [-p] POD [CONTAINER] [flags]
```

and

```
$ k help exec
Execute a command in a container.

Usage:
  kubectl exec POD -c CONTAINER -- COMMAND [args...] [flags]
```

and

```
thockin@freakshow kubernetes hostport-to-service /$ k logs
kube-dns-v4-cf036 -c kube2sky | head -0
Error: unknown shorthand flag: 'c' in -c
Run 'kubectl help' for usage.

thockin@freakshow kubernetes hostport-to-service /$ k logs
kube-dns-v4-cf036 kube2sky | head -0

thockin@freakshow kubernetes hostport-to-service /$ k exec
kube-dns-v4-cf036 kube2sky sh
Cannot run exec command
c2eed8914513c7988dedc091c7a61a6c32961afad3acbedb2f5a63363b376aed in
container df224ba26ff9b22c993593c18c398ccec3fbc8128861dbdfd14d1d1a3bf37670:
[8] System error: exec: "kube2sky": executable file not found in $PATH
Error starting exec command in container
c2eed8914513c7988dedc091c7a61a6c32961afad3acbedb2f5a63363b376aed: Cannot
run exec command
c2eed8914513c7988dedc091c7a61a6c32961afad3acbedb2f5a63363b376aed in
container df224ba26ff9b22c993593c18c398ccec3fbc8128861dbdfd14d1d1a3bf37670:
[8] System error: exec: "kube2sky": executable file not found in $PATH
error: Error executing remote command: Error executing command in
container: Error executing in Docker Container: -1

thockin@freakshow kubernetes hostport-to-service /$ k exec
kube-dns-v4-cf036 -c kube2sky sh

thockin@freakshow kubernetes hostport-to-service /$
```

They are very much different from each other.  I think `logs` needs to grow
a `-c` and/or they both should accept "pod/container" syntax.

**Comment by bgrant0607** [comment]
So you're right.  
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/log.go#L67

We couldn't get rid of the `-c`, so logs should adopt that convention, I guess.

**Comment by thockin** [comment]
Any thoughts on `pod[/container]` as syntax?  That saves -c for actual
meaning and gives us a notation for any other commands that can "figure
out" what container or let users specify one.

**Comment by bgrant0607** [comment]
@thockin I'm not opposed to podname/containername, though that would mean we might need to support pods/podname/containername #8884.

**Comment by thockin** [comment]
We don't want to confuse this CLI syntax with REST syntax - to imply that
`pods/podname/containername` is a valid REST resource would be bad, I think
(we did not namespace for that, e.g. bindings, exec, etc subresources would
collide).

How about `pods/podname[:containername]` as syntax or
`pods/podname[#containername]` or `pods/podname[@containername]`

**Comment by thockin** [comment]
Could any of these possibly work as less-than-ideal REST paths in v1, with
v2 being better handled?

**Comment by bgrant0607** [comment]
The REST pattern is collection/instance, so containers as a nested resource #1900 would be pods/podname/containers/containername. The CLI form could elide "containers", much as we've discussed with namespaces #3731.

**Comment by bgrant0607** [comment]
I'd be fine with colon as a separator for now, in addition to (preferrably) POD CONTAINER or POD -c CONTAINER, if the former can't be made to work for both.

**Comment by JanetKuo** [comment]
Issue fixed with #10519
